jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/cache/CacheStore.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.jena.fuseki.cache;
+
+import org.apache.jena.fuseki.Fuseki;
+import org.apache.jena.fuseki.servlets.ActionLib;
+import org.apache.jena.fuseki.servlets.HttpAction;
+import org.slf4j.Logger;
+
+import javax.servlet.http.HttpServletRequest;
+
+public class CacheStore {
+
+
+    /** flag to check if data store was initialized */
+    public static boolean initialized = false ;
+
+    private static Logger log = Fuseki.cacheLog ;
+
+    /** execution timeout for any method */
+    private int defaultExecutionTimeout = 500 ;
+
+    /** thread pool size for this data store */
+    protected int defaultThreadPoolSize = 500 ;
+
+    /** client for interacting with  Cache store **/
+    private final CacheClient client = new GuavaCacheClient();
+
+    private static CacheStore instance;
+
+    public static CacheStore getInstance(){
+
+        if(instance==null){
+            instance = new CacheStore();
+            return instance;
+        }
+        return instance;
+    }
+
+    /** For testing - reset the places which initialize once */
```

Comment:
Singletons can be tricky, sometimes, because OSGi doesn't offer a single unified classpath to a multi-module application or library. It's not always an issue, but unless you have a really good reason to have a singleton (and I would argue here that you do not-- you are trying to build a single cache per-Fuseki-instance, not per-JVM, and as @afs pointed out, that is probably not the right approach in the end _anyway_) it's better to keep it simple.